### ISSUE TITLE
[Beta] Large queries like Example 1e (without taxon restriction) crashes with "Connection reset"

### ISSUE THREAD

**Issue by sholmas** [BioGateway/cytoscape-app#30]
The query will execute fine when simplified and executed in the endpoint, but not when including the context data we need for the App. This has been a “workaround” in the app for a while, injecting extra data into the query to be returned, to simplify the parsing process, but it might cause issues now.

This “simplified” query works:

```
BASE <http://rdf.biogateway.eu/graph/>
SELECT DISTINCT  ?A ?B ?C
WHERE {
GRAPH <tfac2gene> {
<http://rdf.biogateway.eu/prot/9606/chr-X/FOXO4/UPI00002122FA> <http://purl.obolibrary.org/obo/RO_0002428> ?A .
}
GRAPH <prot2bp> {
?B <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> .
}
GRAPH <gene> {
?A <http://semanticscience.org/resource/SIO_010078> ?B .
}
GRAPH <prot2prot> {
?B <http://purl.obolibrary.org/obo/RO_0002436> ?C .
}
GRAPH <prot2bp> {
?C <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> .
}
FILTER(?A!=?C)
FILTER(?A!=?B)
FILTER(?C!=?B)
}
```

This is the query generated by the app, and crashes:

```
BASE <http://rdf.biogateway.eu/graph/>
SELECT DISTINCT <http://rdf.biogateway.eu/prot/9606/chr-X/FOXO4/UPI00002122FA> as ?rdf_biogateway_eu_prot_9606_chr_X_FOXO4_UPI00002122FA0 <tfac2gene> <http://purl.obolibrary.org/obo/RO_0002428> ?A as ?A0 ?B as ?B1 <prot2bp> <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> as ?purl_obolibrary_org_obo_GO_00016661 ?A as ?A2 <gene> <http://semanticscience.org/resource/SIO_010078> ?B as ?B2 ?B as ?B3 <prot2prot> <http://purl.obolibrary.org/obo/RO_0002436> ?C as ?C3 ?C as ?C4 <prot2bp> <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> as ?purl_obolibrary_org_obo_GO_00016664 
WHERE {
{ SELECT DISTINCT  ?A ?B ?C
WHERE {
GRAPH <tfac2gene> {
<http://rdf.biogateway.eu/prot/9606/chr-X/FOXO4/UPI00002122FA> <http://purl.obolibrary.org/obo/RO_0002428> ?A .
}
GRAPH <prot2bp> {
?B <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> .
}
GRAPH <gene> {
?A <http://semanticscience.org/resource/SIO_010078> ?B .
}
GRAPH <prot2prot> {
?B <http://purl.obolibrary.org/obo/RO_0002436> ?C .
}
GRAPH <prot2bp> {
?C <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> .
}
}}

FILTER(?A!=?C)
FILTER(?A!=?B)
FILTER(?C!=?B)
}
```

The “SELECT DISTINCT …” part injects a lot of extra return data into the query, so that when the results come back, they can easily be parsed line-by-line by the parser, which does not care about what parts of the query was wildcards and what parts that are static entities. The static entities are thus injected as part of the query like:

`<http://rdf.biogateway.eu/prot/9606/chr-X/FOXO4/UPI00002122FA> as ?rdf_biogateway_eu_prot_9606_chr_X_FOXO4_UPI00002122FA0 <tfac2gene> <http://purl.obolibrary.org/obo/RO_0002428> ?A as ?A0`

In short, this could be replaced by “?A”, if the client would remember the query and context around it.

**Comment by sholmas** [comment]
Interesting edge case. 

This query works:

```
BASE <http://rdf.biogateway.eu/graph/>
SELECT DISTINCT <http://rdf.biogateway.eu/prot/9606/chr-X/FOXO4/UPI00002122FA> as ?rdf_biogateway_eu_prot_9606_chr_X_FOXO4_UPI00002122FA0 <tfac2gene> <http://purl.obolibrary.org/obo/RO_0002428> ?A as ?A0 ?B as ?B1 <prot2bp> <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> as ?purl_obolibrary_org_obo_GO_00016661 ?A as ?A2 <gene> <http://semanticscience.org/resource/SIO_010078> ?B as ?B2 ?B as ?B3 <prot2prot> <http://purl.obolibrary.org/obo/RO_0002436> ?C as ?C3 
WHERE {
{ SELECT DISTINCT  ?A ?B ?C
WHERE {
GRAPH <tfac2gene> {
<http://rdf.biogateway.eu/prot/9606/chr-X/FOXO4/UPI00002122FA> <http://purl.obolibrary.org/obo/RO_0002428> ?A .
}
GRAPH <prot2bp> {
?B <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> .
}
GRAPH <gene> {
?A <http://semanticscience.org/resource/SIO_010078> ?B .
}
GRAPH <prot2prot> {
?B <http://purl.obolibrary.org/obo/RO_0002436> ?C .
}
}}
}
```

While this query crashes:
```
BASE <http://rdf.biogateway.eu/graph/>
SELECT DISTINCT <http://rdf.biogateway.eu/prot/9606/chr-X/FOXO4/UPI00002122FA> as ?rdf_biogateway_eu_prot_9606_chr_X_FOXO4_UPI00002122FA0 <tfac2gene> <http://purl.obolibrary.org/obo/RO_0002428> ?A as ?A0 ?B as ?B1 <prot2bp> <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> as ?purl_obolibrary_org_obo_GO_00016661 ?A as ?A2 <gene> <http://semanticscience.org/resource/SIO_010078> ?B as ?B2 ?B as ?B3 <prot2prot> <http://purl.obolibrary.org/obo/RO_0002436> ?C as ?C3 
WHERE {
{ SELECT DISTINCT  ?A ?B ?C
WHERE {
GRAPH <tfac2gene> {
<http://rdf.biogateway.eu/prot/9606/chr-X/FOXO4/UPI00002122FA> <http://purl.obolibrary.org/obo/RO_0002428> ?A .
}
GRAPH <prot2bp> {
?B <http://purl.obolibrary.org/obo/RO_0002331> <http://purl.obolibrary.org/obo/GO_0001666> .
}
GRAPH <gene> {
?A <http://semanticscience.org/resource/SIO_010078> ?B .
}
GRAPH <prot2prot> {
?B <http://purl.obolibrary.org/obo/RO_0002436> ?C .
}
}}
FILTER(?A!=?C)
FILTER(?A!=?B)
FILTER(?C!=?B)
}
```

The extra 
```
FILTER(?A!=?C)
FILTER(?A!=?B)
FILTER(?C!=?B)
```
Causes it to crash, appearently. This should be unrelated to the metadata injected into the query select, but it still has an effect. Perhaps the straw that broke the camel's back.

**Comment by sholmas** [comment]
This seems to have resolved itself with the latest data in the database.

**Comment by sholmas** [comment]
This seems to have resolved itself with the latest data in the database.
